Tests/ProfileRecorderSampleConversionTests/SampleConversionTests.swift

```diff
@@ -0,0 +1,102 @@
+        let actual = line.utf8.attemptFastParseStackFrame()
+        XCTAssertEqual(StackFrame(instructionPointer: 0x1_8fe2_4c08, stackPointer: 0), actual)
+    }
+
+    func testOrder() {
+        let line = #"{"sp":"0x1702a6fe0","ip":"0x18fe24c08"}"#[...]
+        let actual = line.utf8.attemptFastParseStackFrame()
+        XCTAssertEqual(StackFrame(instructionPointer: 0x1_8fe2_4c08, stackPointer: 0), actual)
+    }
+
+    func testExtraFieldsOrder() {
+        let line = #"{"sp":"0x1702a6fe0","unknown-field":"hello","ip":"0x18fe24c08"}"#[...]
+        let actual = line.utf8.attemptFastParseStackFrame()
+        XCTAssertEqual(StackFrame(instructionPointer: 0x1_8fe2_4c08, stackPointer: 0), actual)
+    }
+
+    func testSimpleCase() {
+        let line = #"{"ip":"0x18fe24c08","sp":"0x1702a6fe0"}"#[...]
+        let actual = line.utf8.attemptFastParseStackFrame()
+        XCTAssertEqual(StackFrame(instructionPointer: 0x1_8fe2_4c08, stackPointer: 0), actual)
+    }
+
+    func testIpFirst() {
+        let line = #"{"ip":"0xABCDEF","sp":"0x1702a6fe0"}"#[...]
+        let actual = line.utf8.attemptFastParseStackFrame()
+        XCTAssertEqual(StackFrame(instructionPointer: 0xABCDEF, stackPointer: 0), actual)
+    }
+
+    func testIpLast() {
+        let line = #"{"sp":"0x1234","thread":"42","flags":"0x9","ip":"0xCAFEBABE"}"#[...]
+        let actual = line.utf8.attemptFastParseStackFrame()
+        XCTAssertEqual(StackFrame(instructionPointer: 0xCAFE_BABE, stackPointer: 0), actual)
+    }
+
+    func testIpWithSpaces() {
+        let line = #"  {   "ip" :   "0xDEAD10CC" , "sp":"0x1702a6fe0" }  "#[...]
+        let actual = line.utf8.attemptFastParseStackFrame()
+        XCTAssertEqual(StackFrame(instructionPointer: 0xDEAD_10CC, stackPointer: 0), actual)
+    }
+
+    func testIpMixedFields() {
+        let line = #"{"foo":1,"bar":true,"ip":"0x12345678","baz":[1,2,3]}"#[...]
+        let actual = line.utf8.attemptFastParseStackFrame()
+        XCTAssertEqual(StackFrame(instructionPointer: 0x1234_5678, stackPointer: 0), actual)
+    }
+
+    func testIpEmbeddedInText() {
+        let line = #"{"message":"before","ip":"0xFEEDFACE","after":"something"}"#[...]
+        let actual = line.utf8.attemptFastParseStackFrame()
+        XCTAssertEqual(StackFrame(instructionPointer: 0xFEED_FACE, stackPointer: 0), actual)
+    }
+
+    func testIpUpperLowerMix() {
+        let line = #"{"ip":"0xDeadBeef"}"#[...]
+        let actual = line.utf8.attemptFastParseStackFrame()
+        XCTAssertEqual(StackFrame(instructionPointer: 0xDEAD_BEEF, stackPointer: 0), actual)
+    }
+
+    func testIpOnly() {
+        let line = #"{"ip":"0x11111111"}"#[...]
+        let actual = line.utf8.attemptFastParseStackFrame()
+        XCTAssertEqual(StackFrame(instructionPointer: 0x1111_1111, stackPointer: 0), actual)
+    }
+
+    func testIpWithEscapedString() {
+        let line = #"{"comment":"the ip is \"secret\"","ip":"0xFACEB00C"}"#[...]
```

Comment:
Should we also add tests for when the `ip` field is:
- missing?
- incomplete `{"ip": "0x123`
- has no closing quote (possibly covers the point above)